src/components/blockMenu/permissions.tsx

```diff
@@ -1,81 +1,312 @@
+	const [snackbarVisible, setSnackbarVisible] = useState(false)
+	const { showActionSheetWithOptions } = useActionSheet()
+
 	type VisibilityResult = { updateVisibility: { id: number; public: boolean } }
 	type VisibilityRequest = { blockId: number; public: boolean }
 	const [visibilityResult, setVisibility] = useMutation<VisibilityResult, VisibilityRequest>(UPDATE_VISIBILITY)
 
+	type PermissionResult = { blockById: Permission }
+	type PermissionRequest = { id: number }
+	const [permissionResponse, getPermissions] = useQuery<PermissionResult, PermissionRequest>({
+		query: GET_PERMISSIONS,
+		variables: { id: menu.block_id },
+	})
+
+	type SetPermissionRequest = { blockId: number; full: Array<number>; edit: Array<number>; view: Array<number> }
+	const [, setPermissions] = useMutation<PermissionResult, SetPermissionRequest>(SET_PERMISSIONS)
+
+	useEffect(() => {
+		getPermissions()
+	}, [])
+
+	const permissionsData = [
+		{ title: 'Full', data: full },
+		{ title: 'Edit', data: edit },
+		{ title: 'View', data: view },
+	]
+
 	if (visibilityResult.data?.updateVisibility) {
 		isPublic = visibilityResult.data?.updateVisibility.public
 	}
 
+	useEffect(() => {
+		if (permissionResponse.data?.blockById) {
+			const permissions = permissionResponse.data?.blockById
+			setFull(permissions.full)
+			setEdit(permissions.edit)
+			setView(permissions.view)
+		}
+	}, [permissionResponse])
+
+	// Capturing data from the child component
+	React.useEffect(() => {
+		if (route.params?.user) {
+			const exists = view.some(e => e.id === route.params?.user.id)
+			if (!exists) {
+				setView(oldView => [...oldView, route.params?.user])
+			}
+		}
+	}, [route.params?.user])
+
+	React.useLayoutEffect(() => {
+		navigation.setOptions({
+			headerRight: () => {
+				return (
+					<Appbar.Action
+						icon='plus'
+						onPress={() => {
+							const searchComponent: SearchComponent = {
+								cid: 'search',
+								type: 'User',
+								action_text: 'Select User',
+							}
+							navigation.navigate(routes.SEARCH, { searchComponent: searchComponent, isManualSelection: true })
+						}}
+					/>
+				)
+			},
+		})
+	}, [navigation])
+
 	const handlePermissions = () => {
 		const request: VisibilityRequest = { blockId: menu?.block_id, public: !isPublic }
 		setVisibility(request)
 	}
 
+	const onSubmit = () => {
+		const fullIds = full.map(({ id }: User) => {
+			return id
+		})
+		const editIds = edit.map(({ id }: User) => {
+			return id
+		})
+		const viewIds = view.map(({ id }: User) => {
+			return id
+		})
+
+		const request: SetPermissionRequest = {
+			blockId: menu?.block_id,
+			full: fullIds,
+			edit: editIds,
+			view: viewIds,
+		}
+		setLoading(true)
+		setPermissions(request).then(async ({ data }) => {
+			setLoading(false)
+			if (data != undefined) {
+				setSnackbarVisible(true)
+			}
+		})
+	}
+
+	const showActionSheet = (item: User) => {
+		const options = ['Full', 'Edit', 'View', 'Cancel']
```

Comment:
Can you make it say "Read-Only" instead? Just for the end user.
And also, can you add a "Remove" option?